gallery_dl/extractor/nana.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://nana.my.id/"""
+
+from .common import GalleryExtractor, Extractor, Message
+from .. import text, exception
+import json
+
+
+class NanaGalleryExtractor(GalleryExtractor):
+    """Extractor for image galleries from nana.my.id"""
+    category = "nana"
+    directory_fmt = ("{category}", "{title}")
+    pattern = r"(?:https?://)?nana\.my\.id/reader/([^/?#]+)"
+    test = (
+        (("https://nana.my.id/reader/"
+          "059f7de55a4297413bfbd432ce7d6e724dd42bae"), {
+            "pattern": r"https://nana\.my\.id/reader/"
+                       r"\w+/image/page\?path=.*\.\w+",
+            "title"  : "Everybody Loves Shion",
+            "artist" : "fuzui",
+            "tags"   : list,
+            "count"  : 29,
+        }),
+        (("https://nana.my.id/reader/"
+          "77c8712b67013e427923573379f5bafcc0c72e46"), {
+            "pattern": r"https://nana\.my\.id/reader/"
+                       r"\w+/image/page\?path=.*\.\w+",
+            "title"  : "Lovey-Dovey With an Otaku-Friendly Gyaru",
+            "artist" : "Sueyuu",
+            "tags"   : ["Sueyuu"],
+            "count"  : 58,
+        }),
+    )
+
+    def __init__(self, match):
+        self.gallery_id = match.group(1)
+        url = "https://nana.my.id/reader/" + self.gallery_id
+        GalleryExtractor.__init__(self, match, url)
+
+    def metadata(self, page):
+        title = text.unescape(
+            text.extract(page, '</a>&nbsp; ', '</div>')[0])
+        artist = text.unescape(text.extract(
+            page, '<title>', '</title>')[0])[len(title):-10]
+        tags = text.extract(page, 'Reader.tags = "', '"')[0]
+
+        return {
+            "gallery_id": self.gallery_id,
+            "title"     : title,
+            "artist"    : artist[4:] if artist.startswith(" by") else "",
+            "tags"      : tags.split(", ") if tags else [],
```

Comment:
```suggestion
            "tags"      : tags.split(", ") if tags else (),
```

Saves a list allocation